validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -90,29 +92,47 @@ public void validate() throws Exception {
     RetryHelper.retry(
         maxRetryCount,
         () -> {
-          // We will query both the Service and RemoteService dimensions to ensure we get all
-          // metrics from all aggregations, specifically the [RemoteService] aggregation.
+          // We will query metrics that include Service, RemoteService, or RemoteTarget dimensions
+          // to ensure we get all metrics from all aggregations, specifically the [RemoteService] aggregation.
           List<String> serviceNames =
               Lists.newArrayList(
                   context.getServiceName(), context.getRemoteServiceDeploymentName());
-          // TODO - Put the following back in: "www.amazon.com",  "AWS.SDK.S3"
+          // TODO - Put the following back in: "www.amazon.com"
           List<String> remoteServiceNames =
               Lists.newArrayList(context.getRemoteServiceDeploymentName());
           if (context.getRemoteServiceName() != null && !context.getRemoteServiceName().isEmpty()) {
             serviceNames.add(context.getRemoteServiceName());
           }
 
           List<Metric> actualMetricList = Lists.newArrayList();
-          addMetrics(
-              CloudWatchService.SERVICE_DIMENSION,
-              serviceNames,
-              expectedMetricList,
-              actualMetricList);
-          addMetrics(
-              CloudWatchService.REMOTE_SERVICE_DIMENSION,
-              remoteServiceNames,
-              expectedMetricList,
-              actualMetricList);
+
+          // Add sets of dimension filters to use for each query to CloudWatch.
+          List<List<Pair<String, String>>> dimensionLists = Lists.newArrayList();
+          // Query metrics that includes any of these <service> values.
+          for (String serviceName : serviceNames) {
+            dimensionLists.add(
+                Arrays.asList(new Pair<>(CloudWatchService.SERVICE_DIMENSION, serviceName)));
+          }
+          // Query metrics that includes any of these <remoteService> values.
+          for (String remoteServiceName : remoteServiceNames) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, remoteServiceName)));
+          }
+          // Query for metrics that includes both of these <remoteService, remoteTarget> values.
+          // Querying just 'remoteService="AWS.SDK.S3"' would also work, but that can result in
+          // returning too many canary metrics and may cause issues.
+          if (context.getTestingId() != null && !context.getTestingId().isEmpty()) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, "AWS.SDK.S3"),
```

Comment:
Wouldn't looking for `AWS.SDK.S3` still result in many unwanted metrics? Are we querying that value alongside the bucket name to ensure we don't get unwanted metrics? Just making sure my understanding is correct.

Reply:
Yeah, then as I stated in the reply to the comment thread above this one the `RemoteService == AWS.SDK.S3` is not needed anyways since we're looking for `bucket-name` which is unique.

Perhaps this can help make the code less complicated. Your refactor to use a list of pairs of dimension key-values works well but if we can avoid having such complicated logic it would be best.

Reply:
Removed refactor, now using previous logic to query `bucket-name`.